src/service/ngJwtAuthService.ts

```diff
@@ -721,4 +721,20 @@ export class NgJwtAuthService {
         return this.retrieveAndProcessToken(endpoint, authHeader);
     }
 
+    /**
+     * @returns boolean
+     */
+    public tokenHasExpired():boolean {
+
+        this.tokenData = this.readToken(this.rawToken);
+
+        if (!this.rawToken) {
+            return false;
+        }        
+        
+        let expiryDate = moment(this.tokenData.data.exp * 1000);
+
+        return expiryDate < moment();
+    }
```

Comment:
@rayzor65 first off, hey!

What is the use case for this method? If it is just to expose an expired state to the public contract then the code in `public processNewToken` should be refactored to use this method and avoid duplication.

Otherwise, if it is just handling the exceptional case, then something like

```ts
try {
  return this.authService.processNewToken(token);
} catch (e) {
  if (e instanceof NgJwtAuthTokenExpiredException) {
    // handle expired token case
  }
  throw e;  
}
```
Might be a better pattern for the use case.